search_service/proxy/elasticsearch.py

```diff
@@ -376,19 +262,30 @@ def validate_filter_values(search_request: dict) -> Any:
             return True
 
     @staticmethod
-    def parse_query_term(query_term: str) -> str:
+    def parse_query_term(query_term: str,
+                         index: str) -> str:
         # TODO: Might be some issue with using wildcard & underscore
         # https://discuss.elastic.co/t/wildcard-search-with-underscore-is-giving-no-result/114010/8
-        return f'(name:(*{query_term}*) OR name:({query_term}) ' \
-               f'OR schema:(*{query_term}*) OR schema:({query_term}) ' \
-               f'OR description:(*{query_term}*) OR description:({query_term}) ' \
-               f'OR column_names:(*{query_term}*) OR column_names:({query_term}) ' \
-               f'OR column_descriptions:(*{query_term}*) OR column_descriptions:({query_term}))'
+        if index == TABLE_INDEX:
+            query_term = f'(name:(*{query_term}*) OR name:({query_term}) ' \
+                         f'OR schema:(*{query_term}*) OR schema:({query_term}) ' \
+                         f'OR description:(*{query_term}*) OR description:({query_term}) ' \
+                         f'OR column_names:(*{query_term}*) OR column_names:({query_term}) ' \
+                         f'OR column_descriptions:(*{query_term}*) OR column_descriptions:({query_term}))'
+        elif index == DASHBOARD_INDEX:
+            query_term = f'(name:(*{query_term}*) OR name:({query_term}) ' \
```

Comment:
@feng-tao isn't this piece supposed to search across all the parameters supported from query string search? 

So just like tables is searching across `name`, `schema`, `description`, `column_names` and `column_description` then shouldn't this piece for dashboard search across `name`, `group_name`, `description` and `query_name`?

Reply:
that is for query_term, for filter category, it is defined at https://github.com/lyft/amundsensearchlibrary/blob/a56bea64b39a32416216f3a53290f5260dfe5e49/search_service/proxy/elasticsearch.py#L348-L360 . So if users put filter as well as input query term(put the text in the search bar in the advanced search UI), it will first construct the query DSL + query term which we wrap the query term with wild card to search across different facet.

Reply:
I'm not understanding. In the case where there is a query term and a filter for tables we are searching across the "main" fields above. But here for dashboards we are only searching across the "filter" fields and not applying the query term to things like `description` and `query_name`. 

So if someone entered a query name in the search bar and then wanted to filter those results, what would they get with this implementation?

Reply:
Ah I see, we could change that to apply the same. Thanks for catching that.